model/publishing/delivery/PublishingDeliveryService.php

```diff
@@ -70,68 +66,34 @@ public function publishDelivery(\core_kernel_classes_Resource $delivery)
-        foreach ($environments as $env) {
-            if ($this->checkActionForEnvironment(DeliveryCreatedEvent::class, $env)) {
-                $callBackTask = new CallbackTask($taskLog->getId(), $taskLog->getOwner());
-                $task = $queueDispatcher->createTask(new DeployTestEnvironments(), [$test->getUri(), $env->getUri(), $delivery->getUri()], __('Publishing %s to remote env %s', $delivery->getLabel(), $env->getLabel()), $callBackTask);
-                $message = DeployTestEnvironments::class . "task created; Task id: " . $task->getId();
-                $report->add(\common_report_Report::createSuccess($message));
-                $this->logNotice($message);
-            }
+        $report = \common_report_Report::createInfo('Publishing delivery ' . $delivery->getUri());
+
+        if (empty($environments)) {
+            $report->add(\common_report_Report::createFailure("There is no publication targets to deploy"));
+
+            return $report;
         }
-        return $report;
-    }
 
-    /**
-     * @param \core_kernel_classes_Resource $delivery
-     * @return \common_report_Report
-     * @throws \common_exception_Error
-     * @throws \common_exception_NotFound
-     */
-    public function syncDelivery(\core_kernel_classes_Resource $delivery)
-    {
-        $environments = $this->getEnvironments();
-        /** @var QueueDispatcherInterface $queueDispatcher */
-        $queueDispatcher = $this->getServiceManager()->get(QueueDispatcherInterface::SERVICE_ID);
-        $report = \common_report_Report::createInfo('Updating remote delivery ' . $delivery->getUri());
+        /** @var \core_kernel_classes_Resource $env */
         foreach ($environments as $env) {
-            if ($this->checkActionForEnvironment(DeliveryUpdatedEvent::class, $env)) {
-                $task = $queueDispatcher->createTask(new SyncDeliveryEnvironments(), [$delivery->getUri(), $env->getUri()], __('Updating %s to remote env %s', $delivery->getLabel(), $env->getLabel()));
-                $message = DeployTestEnvironments::class . "task created; Task id: " . $task->getId();
-                $report->add(\common_report_Report::createSuccess($message));
-                $this->logNotice($message);
-            }
+            $callBackTask = new CallbackTask($taskLog->getId(), $taskLog->getOwner());
+            $task = $queueDispatcher->createTask(
+                new DeployTestEnvironments(),
+                [$test->getUri(), $env->getUri(), $delivery->getUri()],
+                __(
+                    'Publishing %s to remote env %s',
+                    $delivery->getLabel(),
+                    $env->getLabel()
+                ),
+                $callBackTask
+            );
+            $message = DeployTestEnvironments::class . "task created; Task id: " . $task->getId();
+            $report->add(\common_report_Report::createSuccess($message));
+            $this->logNotice($message);
         }
         return $report;
```

Comment:
maybe it will be better to add new line before return ?

Reply:
Fixed. :)